vk_video_transcoder/demos/vk-video-trans/Main.cpp

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright (C) 2016 Google, Inc.
+ * Copyright 2020 NVIDIA Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <assert.h>
+#include <string>
+#include <vector>
+#include <cstring>
+#include <cstdio>
+
+#include "VkCodecUtils/VulkanDeviceContext.h"
+#include "VkCodecUtils/ProgramConfig.h"
+#include "VkCodecUtils/VulkanVideoProcessor.h"
+#include "VkCodecUtils/VulkanDecoderFrameProcessor.h"
+#include "VkShell/Shell.h"
+
+#if (_TRANSCODING)
+#include "VkVideoEncoder/VkVideoEncoder.h"
+#include "VkVideoEncoder/VkEncoderConfig.h"
+#include "VkCodecUtils/VulkanVideoDisplayQueue.h"
+#include "VkCodecUtils/VulkanVideoEncodeDisplayQueue.h"
+#include "VkCodecUtils/VulkanEncoderFrameProcessor.h"
+#endif
+
+int main(int argc, const char **argv) {
+
+    ProgramConfig programConfig(argv[0]);
+    programConfig.ParseArgs(argc, argv);
+
+    // In the regular application usecase the CRC output variables are allocated here and also output as part of main.
+    // In the library case it is up to the caller of the library to allocate the values and initialize them.
+    std::vector<uint32_t> crcAllocation;
+    crcAllocation.resize(programConfig.crcInitValue.size());
+    if (crcAllocation.empty() == false) {
+        programConfig.crcOutput = &crcAllocation[0];
+        for (size_t i = 0; i < programConfig.crcInitValue.size(); i += 1) {
+            crcAllocation[i] = programConfig.crcInitValue[i];
+        }
+    }
+
+    static const char* const requiredInstanceLayers[] = {
+        "VK_LAYER_KHRONOS_validation",
+        nullptr
+    };
+
+#if _TRANSCODING
```

Comment:
If this is the main for transcoding I dont see the point to have `#if _TRANSCODING`

Reply:
The point of this `#if _TRANSCODING` is a code maintaining - with that I will have clear understanding what parts in transcoding's main I should update when the decoder's main is updated

Reply:
-> Replacing _TRANCODING by something else like VVS_TRANSCODING
it's OK
-> Other preliminary remark would be to not mix video_decoder and video_transcoder. I would have a separate CMake project for one or the other.
In my opinion adding a little changes under transcoder's ifdef (a few lins) to the decoder's CMake file is better than copying the same code to the transcoder's new CMake file and modifying it there.
-> In a next future it would be nice to share common code in a library to avoid the rebuild.
I think this will be difficult to do because transcoder will be based on the decoder's and encoder's folders. By fact this is the same code (because of _TRANSCODING in CMake, the added code section aren't triggered)